GoogleUtilities/AppDelegateSwizzler/README.md

```diff
@@ -0,0 +1,151 @@
+# App Delegate Swizzler
+
+## Overview
+
+The App Delegate Swizzler swizzles certain methods on the AppDelegate and allows interested parties (for eg. other SDKs like Firebase Analytics) to register listeners when certain App Delegate methods are called. 
+
+The App Delegate Swizzler uses [isa swizzling](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/KeyValueObserving/Articles/KVOImplementation.html) to create a dynamic subclass of the app delegate class and add methods to it that have the logic to add multiple "interceptors".
+
+Adding interceptors to the following methods is currently supported by the App Delegate Swizzler.
+
+* `- (BOOL)application:openURL:options:` [Reference](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623112-application?language=objc)
+
+Note: This method is added only if the original app delegate implements it. See [the code](https://github.com/firebase/firebase-ios-sdk/blob/a7117ea786f4d8f45b798ceb19b75e5400f2899e/GoogleUtilities/AppDelegateSwizzler/GULAppDelegateSwizzler.m#L313) for a complete explanation.
```

Comment:
Any way not to lock to a particular version.

Reply:
Not really - I could copy over the explanation here?

Reply:
Whichever you think is more robust

Reply:
Done.